### PR TITLE
Quadlet - Use = sign when setting the pull arg for build

### LINKED ISSUE #24599: Quadlet [build] generates invalid `Pull` command lines
### Issue Description

When applying `Pull=...` option in `[build]` section via quadlet, the generated command line has the format `--pull ...` format instead of `--pull=...` format, which is supposedly not supported by buildah.

Currently, I have to use `PodmanArgs=--pull=...` to avoid this issue.

### Steps to reproduce the issue

1. Create a `.build` file with `Pull=newer` (newer is a valid pull policy), in rootless podman.
2. Reload systemd user daemon and run `systemctl --user show -pExecStart` for the generated build unit.

### Describe the results you received

The generated `ExecStart` command is `--pull newer` instead of `--pull=newer`, which causes ambiguity in parsing.

### Describe the results you expected

The generated command should be `--pull=newer`.

Since `podman build` also supports `--pull` as a bare key, I am not sure how to support this in a quadlet file.

### podman info output

```yaml
host:
  arch: amd64
  buildahVersion: 1.38.0
  cgroupControllers:
  - cpu
  - memory
  - pids
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: conmon-1:2.1.12-1
    path: /usr/bin/conmon
    version: 'conmon version 2.1.12, commit: e8896631295ccb0bfdda4284f1751be19b483264'
  cpuUtilization:
    idlePercent: 99.82
    systemPercent: 0.06
    userPercent: 0.12
  cpus: 22
  databaseBackend: sqlite
  distribution:
    distribution: arch
    version: unknown
  eventLogger: journald
  freeLocks: 2038
  hostname: hq
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 10000
      size: 1
    - container_id: 1
      host_id: 165536
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 10000
      size: 1
    - container_id: 1
      host_id: 165536
      size: 65536
  kernel: 6.6.61-1-lts
  linkmode: dynamic
  logDriver: journald
  memFree: 124850552832
  memTotal: 134954811392
  networkBackend: netavark
  networkBackendInfo:
    backend: netavark
    dns:
      package: aardvark-dns-1.13.1-1
      path: /usr/lib/podman/aardvark-dns
      version: aardvark-dns 1.13.1
    package: netavark-1.13.0-1
    path: /usr/lib/podman/netavark
    version: netavark 1.13.0
  ociRuntime:
    name: crun
    package: crun-1.18.2-1
    path: /usr/bin/crun
    version: |-
      crun version 1.18.2
      commit: 00ab38af875ddd0d1a8226addda52e1de18339b5
      rundir: /run/user/10000/crun
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +YAJL
  os: linux
  pasta:
    executable: /usr/bin/pasta
    package: passt-2024_10_30.ee7d0b6-1
    version: |
      pasta 2024_10_30.ee7d0b6
      Copyright Red Hat
      GNU General Public License, version 2 or later
        <https://www.gnu.org/licenses/old-licenses/gpl-2.0.html>
      This is free software: you are free to change and redistribute it.
      There is NO WARRANTY, to the extent permitted by law.
  remoteSocket:
    exists: true
    path: /run/user/10000/podman/podman.sock
  rootlessNetworkCmd: pasta
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /etc/containers/seccomp.json
    selinuxEnabled: false
  serviceIsRemote: false
  slirp4netns:
    executable: ""
    package: ""
    version: ""
  swapFree: 17179865088
  swapTotal: 17179865088
  uptime: 23h 57m 55.00s (Approximately 0.96 days)
  variant: ""
plugins:
  authorization: null
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  - ipvlan
  volume:
  - local
registries: {}
store:
  configFile: ~/.config/containers/storage.conf
  containerStore:
    number: 9
    paused: 0
    running: 9
    stopped: 0
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: ~/.local/share/containers/storage
  graphRootAllocated: 3838601478144
  graphRootUsed: 321813139456
  graphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Supports shifting: "false"
    Supports volatile: "true"
    Using metacopy: "false"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 23
  runRoot: /run/user/10000/containers
  transientStore: false
  volumePath: ~/.local/share/containers/storage/volumes
version:
  APIVersion: 5.3.0
  Built: 1731771105
  BuiltTime: Sat Nov 16 09:31:45 2024
  GitCommit: 874bf2c301ecf0ba645f1bb45f81966cc755b7da
  GoVersion: go1.23.3
  Os: linux
  OsArch: linux/amd64
  Version: 5.3.0
```

### Podman in a container

No

### Privileged Or Rootless

Rootless

### Upstream Latest Release

Yes

### Additional environment details

Additional environment details

### Additional information

Additional information like issue happens only occasionally or issue happens with a particular architecture or on a particular setting

### PR DESCRIPTION
**Pull request by ygalblum**
#### Does this PR introduce a user-facing change?
No

```release-note
None
```

Fixes #24599

### CODE REVIEW

File: test/system/252-quadlet.bats [hunk 1]
```diff
@@ -1699,4 +1699,31 @@ EOF
...
+FROM $IMAGE
+EOF
+
+    local image_tag=quay.io/i-$(safename):$(random_string)
+    local quadlet_file=$PODMAN_TMPDIR/pull_$(safename).build
+    cat >$quadlet_file << EOF
+[Build]
+ImageTag=$image_tag
+File=$container_file_path
+Pull=always
```
Comment (Luap99) [inline test/system/252-quadlet.bats:1721]: Can you set this to `never` , which policy should not matter and never is faster and less flaky as we do not have to pull

Comment (ygalblum) [reply test/system/252-quadlet.bats]: Done

**Review by Luap99 — APPROVED** [review]
LGTM

**Comment by Luap99** [comment]
A bit to late for 5.3.1 unfortunately given it was a 5.3 regression but we include it in the next one. cc @mheon
I guess this one is not so bad given users can work around it as mentioned in the issue so a new release is not time critical.

/cherry-pick v5.3

File: test/system/252-quadlet.bats [hunk 2]
```diff
@@ -1699,4 +1699,31 @@ EOF
         fi
     done < <(parse_table "${dropin_files}")
 }
+
+# Following issue: https://github.com/containers/podman/issues/24599
+# Make sure future changes do not break
+@test "quadlet - build with pull" {
```
Comment (edsantiago) [inline test/system/252-quadlet.bats:1705]: This is going to cause flakes. `buildah` cannot run in parallel, hence `podman build` cannot run in parallel: https://github.com/containers/buildah/issues/5674

In my foolish youth, I overoptimistically added the `ci:parallel` tag to this entire file. That will apply here, which will cause this test to flake. 

Proposed solution(?): override `file_tags`, or move this test into a separate file. I don't know which is better. To override, [it looks like you can redefine file_tags](https://bats-core.readthedocs.io/en/stable/writing-tests.html#tagging-tests), then redefine it again below this test. That's barfy. So is a separate file. There might be better solutions that I can't think of right now.

Comment (Luap99) [reply test/system/252-quadlet.bats:1705]: Will this cause flakes even where there no layers? The Containerfile is just `FROM xxx` so is this already enough to trigger flakes?

Comment (edsantiago) [reply test/system/252-quadlet.bats:1705]: I honestly don't know. I've never dived deeply enough into that failure. Maybe it's OK to leave this as it is, and play the wait-and-see game.

**Comment by mheon** [comment]
We can plan on a 5.3.2 in two weeks, given next week is a major US holiday.

File: test/system/252-quadlet.bats (hunk 2 above)
Comment (ygalblum) [reply test/system/252-quadlet.bats:1705]: @edsantiago I rather not have to split the files. But, I don't mind clearing the flag and setting it again around the test. Do you think it's safer? No point in running later to fix it

Comment (edsantiago) [reply test/system/252-quadlet.bats:1705]: I'm really sorry... I don't know. I've never actually tried the change-file-tags trick. It seems worth a try. It also might be okay to merge as-is and see if it flakes or not: that might provide insight into the nature of the buildah-parallel bug.

Comment (ygalblum) [reply test/system/252-quadlet.bats:1705]: So, if you're OK with it, I'll keep it this way and we'll see what happens. Thanks

**Comment by ygalblum** [comment]
The windows installer failures don't seem related to this change. Is it a known consistent issue, or do I need to retry it until it's successful?

**Comment by edsantiago** [comment]
I've pressed re-run. Let's see.

**Comment by TomSweeneyRedHat** [comment]
LGTM

**Comment by TomSweeneyRedHat** [comment]
I've pressed the test retry button.  When I see "failed" tests with little to no log output, it's generally a try again situation.

**Comment by rhatdan** [comment]
Anyone know why the windows tests are blowing up?  Iwould force merge this, but I am not allowed to, since the Windows failures have nothing to do with this PR, as far as I can tell.

**Comment by Luap99** [comment]
The (temporary) win installer fix is in https://github.com/containers/podman/pull/24653

**Comment by ygalblum** [comment]
@Luap99 @rhatdan I rebased. Hopefully the tests pass

**Pull request merged**